src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -867,6 +935,50 @@ mod tests {
         assert!(pool.get_filesystem_by_uuid(fs_uuid).is_none());
     }
 
+    #[test]
+    pub fn loop_test_meta_expand() {
+        // This test requires more than 1 GiB.
+        loopbacked::test_with_spec(loopbacked::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    #[test]
+    pub fn real_test_meta_expand() {
+        real::test_with_spec(real::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    fn test_meta_expand(paths: &[&Path]) -> () {
+        let pool_uuid = Uuid::new_v4();
+        let dm = DM::new().unwrap();
+        let mut mgr = BlockDevMgr::initialize(pool_uuid, paths, MIN_MDA_SECTORS, false).unwrap();
```

Comment:
Right here, you could be passing paths which are not the original device paths, but are actually a linear device or devices that is only a little bit bigger than the initial size of the pool. Then it is still necessary to figure out how to expand the metadata faster than the data. Possibly you could do that by making lots and lots of filesystems of negligible size.

Reply:
How will providing a smaller amount of storage to the pool do any good?  Doing that won't change the initial size of the meta or the data devices.

Reply:
You're correct. I had the problem backwards, unfortunately.